src/System.Linq/tests/OrderByDescendingTests.cs

```diff
@@ -0,0 +1,171 @@
+
+            Assert.Equal(expected, source.OrderByDescending(e => e));
+        }
+
+        [Fact]
+        public void KeySelectorCalled()
+        {
+            Record[] source = new Record[]
+            {
+
+                new Record{ Name = "Alpha", Score = 90 },
+                new Record{ Name = "Robert", Score = 45 },
+                new Record{ Name = "Prakash", Score = 99 },
+                new Record{ Name = "Bob", Score = 0 }
+            };
+            Record[] expected = new Record[]
+            {
+                new Record{ Name = "Robert", Score = 45 },
+                new Record{ Name = "Prakash", Score = 99 },
+                new Record{ Name = "Bob", Score = 0 },
+                new Record{ Name = "Alpha", Score = 90 }
+            };
+
+            Assert.Equal(expected, source.OrderByDescending(e => e.Name, null));
+        }
+
+        [Fact]
+        public void FirstAndLastAreDuplicatesCustomComparer()
+        {
+            string[] source = { "Prakash", "Alpha", "DAN", "dan", "Prakash" };
+            string[] expected = { "Prakash", "Prakash", "DAN", "dan", "Alpha" };
+
+            Assert.Equal(expected, source.OrderByDescending(e => e, StringComparer.OrdinalIgnoreCase));
+        }
+
+        [Fact]
+        public void FirstAndLastAreDuplicatesNullPassedAsComparer()
+        {
+            int[] source = { 5, 1, 3, 2, 5 };
+            int[] expected = { 5, 5, 3, 2, 1 };
+
+            Assert.Equal(expected, source.OrderByDescending(e => e, null));
+        }
+
+        [Fact]
+        public void SourceReverseOfResultNullPassedAsComparer()
+        {
+            int[] source = { -75, -50, 0, 5, 9, 30, 100 };
+            int[] expected = { 100, 30, 9, 5, 0, -50, -75 };
+
+            Assert.Equal(expected, source.OrderByDescending(e => e, null));
+        }
+
+        [Fact]
+        public void SameKeysVerifySortStable()
+        {
+            Record[] source = new Record[]
+            {
+                new Record{ Name = "Alpha", Score = 90 },
+                new Record{ Name = "Robert", Score = 45 },
+                new Record{ Name = "Prakash", Score = 99 },
+                new Record{ Name = "Bob", Score = 90 },
+                new Record{ Name = "Thomas", Score = 45 },
+                new Record{ Name = "Tim", Score = 45 },
+                new Record{ Name = "Mark", Score = 45 },
+            };
+            Record[] expected = new Record[]
+            {
+                new Record{ Name = "Prakash", Score = 99 },
+                new Record{ Name = "Alpha", Score = 90 },
+                new Record{ Name = "Bob", Score = 90 },
+                new Record{ Name = "Robert", Score = 45 },
+                new Record{ Name = "Thomas", Score = 45 },
+                new Record{ Name = "Tim", Score = 45 },
+                new Record{ Name = "Mark", Score = 45 },
+            };
+
+            Assert.Equal(expected, source.OrderByDescending(e => e.Score));
```

Comment:
ThenBy was removed to make the test actually check stability.

To judge from the comments, I believe what happened was:

1. The test was equivalent to how I have it now. (The equivalent is included and commented out in the old file), and tested stability.
2. When adopted for PLINQ, the test failed, which is fine as PLINQ's OrderBy doesn't promise stability.
3. Rather than just remove this test, it was changed to coerce the results into a stable order by imposing a total ordering outside of `OrderBy` with a combination of Select and ThenBy. At this point the results were stable, but the test code made them stable rather than the tested code, so linq-to-object's OrderBy's stability (which is specified) was no longer tested for.
4. Due to problems with other PLINQ code not being stable in ordering, `Verification.Allequal` was changed to be order-insensitive, making most of these tests pointless anyway. (Also why the earlier version of what I call `FirstAndLastAreDuplicatesNullPassedAsComparer` wasn't failing; it wasn't really checked).

Removing the `Select` and `ThenBy` turns this back into a test for stability.